### DISCUSSION TITLE
Pre-trained Transducer Model

### DISCUSSION THREAD

**Discussion by srikanthk20** [espnet/espnet#3808] [General]
Hi,
Is there a pretrained Transducer model (any variant of RNN, Transformer or Conformer) available to get started.
I checked the Zoo and the Hugging face repos and couldn't find one.
If its pretrained on librspeech its even better.
Thanks in advance

**Comment by sw005320** [comment]
We don't support a pretrained Transducer model in espnet model zoo, but you can find some of them from our Google drive (see the RESULTS.md file. You can find some of them).

**Reply by srikanthk20** [reply]
@sw005320 ... Given your vast experience need to pick your brain on the following. Hope its fine
1. If we have to pick a ASR model that supports both the online (Streaming) & offline service then what do you suggest
2. Do you suggest the transducer or transformer approach
This is for American english and not on-device
Any information will be helpful

**Reply by sw005320** [reply]
In my opinion, both approaches would have pros. and cons.
We tried to minimize the model change for offline and online as much as possible, including the performance gap, and find a current solution (blcokwise online transformer).
We can do the same effort for transducer but we don't provide such a solution in the current repo.

**Reply by srikanthk20** [reply]
@sw005320 ... thanks for the info. Which recipe should i refer to get started on "blockwise online transformer" and the reference papers corresponding to the implementation.

**Reply by sw005320** [reply]
@D-Keqi, can you introduce your activities?
Also, can you add more documentations (e.g., in main README.md and https://espnet.github.io/espnet/tutorial.html)?

**Reply by D-Keqi** [reply]
Sure! @sw005320

**Reply by D-Keqi** [reply]
Hi! @srikanthk20 We now have an example on AISHELL-1 https://github.com/espnet/espnet/blob/master/egs2/aishell/asr1/run_streaming.sh, and there is also an English example on Tedlium2 that has not been merged https://github.com/D-Keqi/espnet/blob/my_streaming_transformer/egs2/tedlium2/asr1/run_streaming.sh
And for reference paper: https://arxiv.org/pdf/2006.14941.pdf

**Reply by srikanthk20** [reply]
Thanks @D-Keqi ... this is helpful
